### DISCUSSION TITLE
Connecting the devcontainer to the host network

### DISCUSSION THREAD

**Discussion by OmerAvra** [devcontainers/community#83] [Q&A]
Anyone have an idea how to connect the devcontainer to the host network?

Running ICMP pig works fine locally 
<img width="1282" alt="image" src="https://github.com/devcontainers/community/assets/75276256/75f62a97-5d7b-43af-914d-bba1b0039356">

On linux, building a maing.go file and giving it setcap cap_net_raw=+ep  also gets it to work but I am looking for a solutution for this on a container eventually..

Network is unreachable from within
<img width="651" alt="image" src="https://github.com/devcontainers/community/assets/75276256/5b64e350-c02e-402a-8b6b-74fc9830e407">
<img width="367" alt="image" src="https://github.com/devcontainers/community/assets/75276256/9183354a-3a25-4d84-ab3f-e829b40feb9c">

tried a bunch of stuff including messing around with the devcontainer.json
e.g. runArgs": [
    "--network=host",
]

**Comment by samruddhikhandale** [comment]
Hi 👋 

The `--network=host` option should work on Linux, but it might not be enough if your application needs to use raw sockets or other privileged capabilities. You can try adding linux capabilities which will give the container the necessary privileges as well as you can try setting the `privileged` mode. 

Here's how you can modify your `devcontainer.json`: 👇 

```jsonc
{
    "image": "your-image-name",
    "privileged": true,
    "runArgs": ["--network", "host"],
    "capAdd": ["NET_RAW", "NET_ADMIN"]
}
```

Please note that these options increase the privileges of the container, which can have security implications. Make sure you understand the implications and trust the code running inside the container.
